runelite-client/src/main/java/net/runelite/client/plugins/fairyring/FairyRingPlugin.java

```diff
@@ -339,6 +398,17 @@ else if (!Strings.isNullOrEmpty(w.getText()))
 			if (!hidden)
 			{
 				y += c.getDescription().getHeight() + ENTRY_PADDING;
+
+				if (c.getFavorite() != null && c.getFavorite().getSpriteId() == SpriteID.FAIRY_RING_REMOVE_FAVOURITE)
+				{
+					separator.setHidden(false);
+					lastFavorite = c;
+				}
```

Comment:
We should only show the separator if both favorite and non-favorite entries are shown. As-is, you can find only favorited entries and have a trailing separator afterwards, which doesn't look good.
![Screenshot_20240714_110157](https://github.com/user-attachments/assets/0dee1464-b465-44a2-9ba4-ea87e93b899f)

I think this is fixed by un-hiding the separator in the "add padding to the first widget after the separator" case